root/static/scripts/edit/components/FormRowTextList.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * @flow strict
+ * Copyright (C) 2024 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React, {useState} from 'react';
+
+import AddButton from './AddButton.js';
+import FieldErrors from './FieldErrors.js';
+import FormLabel from './FormLabel.js';
+import FormRow from './FormRow.js';
+import RemoveButton from './RemoveButton.js';
+
+component TextListRow(
+  name: string,
+  onChange: (event: SyntheticEvent<HTMLInputElement>) => void,
+  onRemove: (event: SyntheticEvent<HTMLInputElement>) => void,
+  removeButtonLabel: string,
+  value: string,
+) {
+  return (
+    <div className="text-list-row">
+      <input
+        className="value with-button"
+        name={name}
+        onChange={onChange}
+        type="text"
+        value={value}
+      />
+      <RemoveButton onClick={onRemove} title={removeButtonLabel} />
+    </div>
+  );
+}
+
+const initialRows = (repeatable: RepeatableFieldT<FieldT<string>>) => {
+  if (repeatable.field.length === 0) {
+    return [{name: repeatable.html_name + '.0', value: ''}];
+  }
+
+  return repeatable.field.map((field, index) => ({
+    name: repeatable.html_name + '.' + index,
+    value: field.value ?? '',
+  }));
```

Comment:
I don't think we should convert the repeatable field into a simplified `{name, value}` object, rather we should just store the repeatable itself in state. We already have many utilities in root/static/scripts/edit/utility/createField.js and root/static/scripts/edit/utility/pushField.js to help with managing form state.